src/Symfony/Component/ExpressionLanguage/Lexer.php

```diff
@@ -40,11 +40,11 @@ public function tokenize(string $expression)
                 continue;
             }
 
-            if (preg_match('/[0-9]+(?:\.[0-9]+)?([Ee][\+\-][0-9]+)?/A', $expression, $match, 0, $cursor)) {
+            if (preg_match('/((?:[0-9]+(?:_[0-9]+)*|(?:(?:[0-9]+(?:_[0-9]+)*)?\.[0-9]+(?:_[0-9]+)*|[0-9]+(?:_[0-9]+)*\.(?:[0-9]+(?:_[0-9]+)*)?))[eE][+-]?[0-9]+(?:_[0-9]+)*|(?:(?:[0-9]+(?:_[0-9]+)*)?\.[0-9]+(?:_[0-9]+)*|[0-9]+(?:_[0-9]+)*\.(?:[0-9]+(?:_[0-9]+)*)?)|[0-9]+(?:_[0-9]+)*)/A', $expression, $match, 0, $cursor)) {
```

Comment:
BTW this regex was computed from this comment in the PHP source code: https://github.com/php/php-src/blob/172d469b7bc351efbcdf54052be3de289b2fdd62/Zend/zend_language_scanner.l#L1359 which is also in the PHP doc.